coreUtilities/src/main/java/org/opensearch/migrations/transformation/entity/Index.java

```diff
@@ -0,0 +1,7 @@
+package org.opensearch.migrations.transformation.entity;
+
+/**
+ * Represents an Index object for transformation
+ */
+public interface Index extends Entity {
```

Comment:
I know that we're under an opensearch namespace right now, but very little is opensearch specific and there could be other users of a traffic-replay tool and by extension these common coreUtilities.  Longer term, I suspect that  but I suspect you'll want to have a coreTransforms package and some specific ones for ES/OS and then maybe even for specific versions.